sdk/nodejs/runtime/resource.ts

```diff
@@ -306,11 +306,32 @@ async function getResourceURNs(resources: Set<Resource>) {
  * Recursively walk the resources passed in, returning them and all resources reachable from
  * [Resource.__childResources]
  */
-function getTransitivelyReferencedChildResources(resources: Set<Resource>) {
-    // Recursively walk the dependent resources through their children, adding them to the result set.
-    const result = new Set<Resource>();
-    addTransitivelyDependentResources(resources, result);
-    return result;
+function getTransitivelyReferencedChildResources(resources: Set<Resource>, initialResource: Resource) {
+    // First, just recursively walk the dependent resources through their children, adding them to
+    // the result set.
+    const temp = new Set<Resource>();
+    addTransitivelyDependentResources(resources, temp);
```

Comment:
Hm. Looks like I missed earlier that all child resources now depend on all of their sibling resources... That seems pretty bad if it is in fact the case, as it will effectively serialize the children of a component.

Reply:
Though I suppose this only happens when a child resource's parent appears in its dependencies, so this could be quite rare in practice.